query/src/pipelines/transforms/transform_window_func.rs

```diff
@@ -0,0 +1,568 @@
+                                            current_value.as_f64().unwrap() + following as f64,
+                                        );
+                                        x.cmp(&max) == Ordering::Less
+                                    });
+                                    start += offset;
+                                }
+                                _ => (),
+                            }
+                            match frame_end {
+                                WindowFrameBound::Preceding(Some(preceding)) => {
+                                    let offset = order_by_values[start..end].partition_point(|x| {
+                                        let min = DataValue::from(
+                                            current_value.as_f64().unwrap() - preceding as f64,
+                                        );
+                                        x.cmp(&min) != Ordering::Greater
+                                    });
+                                    end = start + offset;
+                                }
+                                WindowFrameBound::CurrentRow => {
+                                    let offset = order_by_values[start..end].partition_point(|x| {
+                                        let peer = DataValue::from(current_value.as_f64().unwrap());
+                                        x.cmp(&peer) != Ordering::Greater
+                                    });
+                                    end = start + offset;
+                                }
+                                WindowFrameBound::Following(Some(following)) => {
+                                    let offset = order_by_values[start..end].partition_point(|x| {
+                                        let max = DataValue::from(
+                                            current_value.as_f64().unwrap() + following as f64,
+                                        );
+                                        x.cmp(&max) != Ordering::Greater
+                                    });
+                                    end = start + offset;
+                                }
+                                _ => (),
+                            }
+                            start..end
+                        })
+                        .collect::<Vec<_>>()
+                }
+            },
+        }
+    }
+
+    /// Actually, the computation is row based
+    fn new_partition_aggr_func(
+        func: AggregateFunctionRef,
+        arguments: &[ColumnRef],
+        arena: Arc<Bump>,
+    ) -> SegmentPoint<StateAddr, Agg> {
+        let rows = arguments[0].len();
+        let state_per_tuple = (0..rows)
+            .map(|i| {
+                arguments
+                    .iter()
+                    .map(|c| c.slice(i, 1))
+                    .collect::<Vec<ColumnRef>>()
+            })
+            .map(|args| {
+                let place = arena.alloc_layout(func.state_layout());
```

Comment:
we should call drop_states to destroy the inner allocation  as `PartialAggregator` did.